File: mobile/lib/services/forum_service.dart
import 'package:http/http.dart' as http;
import 'dart:convert';

class CommunityForumService {
  final String apiBaseUrl;

  CommunityForumService(this.apiBaseUrl);

  Future<dynamic> createPost(Map<String, dynamic> postData) async {
    final response = await http.post(
      Uri.parse('$apiBaseUrl/forum/posts'),
      headers: {'Content-Type': 'application/json'},
      body: json.encode(postData),
    );
    return json.decode(response.body);
  }

  Future<dynamic> getPosts() async {
    final response = await http.get(Uri.parse('$apiBaseUrl/forum/posts'));
    return json.decode(response.body);
  }
}

File: frontend/src/services/forumService.js
import api from './api';

export const createPost = async (postData) => {
  try {
    const response = await api.post('/forum/posts', postData);
    return response.data;
  } catch (error) {
    console.error('Failed to create post:', error);
    throw new Error(error.response?.data?.message || 'Failed to create post');
  }
};

export const getPosts = async () => {
  try {
    const response = await api.get('/forum/posts');
    return response.data;
  } catch (error) {
    console.error('Failed to fetch posts:', error);
    throw new Error(error.response?.data?.message || 'Failed to load posts');
  }
};
